src/main/java/org/trigon/commands/GearRatioCalculationCommand.java

```diff
@@ -1,60 +1,93 @@
-    private final LoggedDashboardBoolean shouldMoveClockwise;
-
+    
     private double startingRotorPosition;
     private double startingEncoderPosition;
     private double gearRatio;
-
-    public GearRatioCalculationCommand(DoubleSupplier rotorPositionSupplier, DoubleSupplier encoderPositionSupplier, DoubleConsumer runGearRatioCalculation, SubsystemBase requirement) {
+    private double startTime;
+    private boolean hasSetStartingPositions = false;
+
+    /**
+     * Creates a new GearRatioCalculationCommand.
+     *
+     * @param rotorPositionSupplier             a supplier that returns the current position of the rotor
+     * @param encoderPositionSupplier           a supplier that returns the current position of the encoder
+     * @param runGearRatioCalculation           a consumer that drives the motor with a given voltage
+     * @param backlashAccountabilityTimeSeconds the time to wait before setting the starting positions in order to account for backlash
+     * @param requirement                       the subsystem that this command requires
+     */
+    public GearRatioCalculationCommand(
+            DoubleSupplier rotorPositionSupplier,
+            DoubleSupplier encoderPositionSupplier,
+            DoubleConsumer runGearRatioCalculation,
+            double backlashAccountabilityTimeSeconds,
+            SubsystemBase requirement) {
         this.rotorPositionSupplier = rotorPositionSupplier;
         this.encoderPositionSupplier = encoderPositionSupplier;
         this.runGearRatioCalculation = runGearRatioCalculation;
         this.subsystemName = requirement.getName();
+        this.backlashAccountabilityTimeSeconds = backlashAccountabilityTimeSeconds;
 
         this.movementVoltage = new LoggedDashboardNumber("GearRatioCalculation/" + this.subsystemName + "/Voltage", 1);
-        this.shouldMoveClockwise = new LoggedDashboardBoolean("GearRatioCalculation/" + this.subsystemName + "/ShouldMoveClockwise", false);
 
         addRequirements(requirement);
     }
 
     @Override
     public void initialize() {
-        startingRotorPosition = rotorPositionSupplier.getAsDouble();
-        startingEncoderPosition = encoderPositionSupplier.getAsDouble();
+        startTime = Timer.getFPGATimestamp();
     }
 
     @Override
     public void execute() {
-        runGearRatioCalculation.accept(movementVoltage.get() * getRotationDirection());
-        gearRatio = calculateGearRatio();
-
-        Logger.recordOutput("GearRatioCalculation/" + subsystemName + "/RotorDistance", getRotorDistance());
-        Logger.recordOutput("GearRatioCalculation/" + subsystemName + "/EncoderDistance", getEncoderDistance());
-        Logger.recordOutput("GearRatioCalculation/" + subsystemName + "/GearRatio", gearRatio);
+        runGearRatioCalculation();
+        if (Timer.getFPGATimestamp() - startTime > backlashAccountabilityTimeSeconds && !hasSetStartingPositions)
+            setStartingPositions();
+        log();
```

Comment:
1. Logging before you set the starting positions isn't good. Log only after you set them.
2. It doesn't seem like you're updating the gear ratio anywhere. 
3. If you're logging only after the backlash time, you should probably set everything to zero initially to avoid problems.

Reply:
I calculate the gear ratio in the end, You want me to update it periodically?

Reply:
yes